torch/csrc/jit/mobile/module.cpp

```diff
@@ -145,8 +145,7 @@ std::string Module::getCallStack(const int64_t debug_handle) const {
 // We really need to change this part, so in the next step for profiling support
 // for delegates, the first thing will be to rewrite how profiling is done
 // for lite interpreter.
-std::string Module::get_forward_method_debug_info(size_t pc) const {
-  auto debug_handle = find_method("forward")->get_debug_handle(pc);
+std::string Module::get_forward_method_debug_info(int64_t debug_handle) const {
```

Comment:
So, if the feature is enabled but the model doesn't contain debug_handle then we return the "Debug info for handle..." right?

Reply:
Yes thats right. 

So this API assumes a valid debug handle is passed. You can potentially pass a random handle and if the model contained debug information, it will just symbolicate that random handle. Not that this behavior is invalid but just pointing it out.
The context where this becomes relevant is that events log "pc" and not debug handle. In that case if you call `get_forward_method_debug_info` with pc you may just get wrong debug information. So there is also `hasDebugHandles` API which tells you that what events logged is debug handle, somewhat indirectly, and not pc. So the way use the API then is:
1. Check `hasDebugHandles` == true
2. Use the handle in events with `get_forward_method_debug_info`.

I dont quite like the disjoint nature of the two, but the reason they exist is to provide meaningful profiling information even when debug handles are not present. And in that case we tag events with PC.

I do think however that saving debug handles always, maybe unconditionally, will be more consistent at the cost always saving debug handles. Downside is that 1) debug handles now have to part of program definition and 2) older models without debug handles dont get meaningful profiling information.

Reply:
yeah, I think we need to evaluate the cost of having debug handles since in some cases (embedded) it may be prohibitive. Ideally, we would make it a pluggable component (I guess it kind of is with the ifdef) so it can be entirely disabled, which may not be a bad idea since doesn't it assume file system access even when you use pc for logging (which won't work on embedded)?

So, I would see just the 2 options:
* With debug handles (at build time) to get the associated profiling/debug features.
* Without debug handles and you won't get any of the features nor the overhead.

Not for this PR but what do you think? We would still have to evaluate the cost of always adding debug handles (I thought you had measured it already?).

Reply:
> With debug handles (at build time) to get the associated profiling/debug features.

Currently this feature is supported via 1) save debug handles and 2) build with eager symbolication.

> Without debug handles and you won't get any of the features nor the overhead.

Similarly this feature is available via 1) not save debug handles and 2) not build with symbolication source code.

I think the options that you propose make sense. My only concern is providing the support for the 2 options via single switch. Given that model training/export pipeline is quite separate from deployment, we will continue to have to provide two switches, 1 and 2 above.

> We would still have to evaluate the cost of always adding debug handles (I thought you had measured it already?).

Yes we did that internally in terms of its impact on the model size, which corresponds to 1 above. For 2 we have not evaluated the impact on code size yet.

> Then, for older models we would probably just not log anything.

Right. So this would just look like how current PR responds to profiling when `hasDebugHandle` returns false, right?